### DISCUSSION TITLE
OpenBSD Installation

### DISCUSSION THREAD

**Discussion by kd4e** [wintercms/winter#982] [Q&A]
A Search on "openbsd" didn't return any discussions here.

I have a fresh OpenBSD 7.3 installation on a Dell OptiFlex 7050 which I want to use as my personal server.

It will host several thousand documents, some with images, as well as web sites, weather station data, and a blog.

Has anyone installed WinterCMS on OpenBSD?

Is there a paint-by-numbers set of instructions to do so - avoiding missing a step and making a mess, please?

Thanks.

**Comment by mjauvin** [comment]
I haven't, but as long as your install meets the requirements, it should work:

https://wintercms.com/docs/v1.2/docs/setup/installation

**Reply by kd4e** [reply]
Does this "Under the hood, the web installer uses Composer **to process and install the dependencies necessary** to run Winter CMS" mean that the Web installer will install any of these dependencies that are not already installed (and upgrade any that require it)?
    cURL
    GD
    Mbstring
    OpenSSL
    PDO (including the drivers of your database server)
    SimpleXML
    ZipArchive

I just found that pkg_info doesn't find Mbstring, PDO, SimpleXML, & ZipArchive - are they available for OpenBSD 7.3, please?

Is this the **MBString** it needs? [https://man.openbsd.org/ASN1_mbstring_copy.3]
Will **PHP-PDO** be installed with PHP from here [](https://cdn.openbsd.org/pub/OpenBSD/7.3/packages/amd64/) - or will the version that matches the installed version of PHP need to be installed separately?

Note: I did find this old set of instructions ... 
```
#pkg_add [ftp://ftp.openbsd.org/pub/OpenBSD/5.4/packages/amd64/php-pdo_mysql-5.3.2...](ftp://ftp.openbsd.org/pub/OpenBSD/5.4/packages/amd64/php-pdo_mysql-5.3.27.tgz)
#ln -sf /etc/php-5.3.sample/pdo_mysql.ini /etc/php-5.3/pdo_mysql.ini
```
I'm guessing that his is what I need for **SimpleXML**? [p5-XML-Simple-2.25p0.tgz](https://cdn.openbsd.org/pub/OpenBSD/7.3/packages/amd64/p5-XML-Simple-2.25p0.tgz]
There are multiple packages with "zip" in their names - it's unclear if any of the listed ones, e.g. this, will do ... php-zip-8.2.3.tgz [https://cdn.openbsd.org/pub/OpenBSD/7.3/packages/amd64/php-zip-8.2.3.tgz].

**Reply by LukeTowers** [reply]
> Does this "Under the hood, the web installer uses Composer to process and install the dependencies necessary to run Winter CMS" mean that the Web installer will install any of these dependencies that are not already installed (and upgrade any that require it)?

No. Composer is a PHP package manager, the dependencies listed by the documentation are all extensions to PHP itself, all of which should come standard in a normal PHP installation. The dependencies mentioned in this paragraph are referring to the PHP libraries & packages that Winter uses (i.e. Laravel, Symfony, Doctrine, etc - the underlying frameworks that Winter is built on top of).

https://computingforgeeks.com/how-to-install-php-8-on-freebsd-system/ looks like it covers most of it, although I'd recommend installing PHP 8.2 now instead of 8.0

> #ln -sf /etc/php-5.3.sample/pdo_mysql.ini /etc/php-5.3/pdo_mysql.ini
PHP 5.3 is beyond ancient and most definitely will not work with Winter. If you're installing something that old you're doing something wrong.

Hopefully that helps, let me know if you have any other questions!

**Reply by kd4e** [reply]
Installing php reported capstone, libsodium, onigurama, femail, femail-chroot, and argon.
Locate finds /usr/share/man/man3/ASNI_mbstring_copy.3 - does that imply that mbstring was installed with the PHP istallation, somewhere?
Locate also finds a gazillion "openssl" files.
I just added php-gd-8.2.11, php-xml-8.2.11, and php-curl-8.2.11
Locate finds a bunch of php-pdo-mysql files - do they meet the needed "pdo" prerequisite?
I appreciate your patient assistance - I'm very new to BSD and to building a server from scratch.

**Reply by kd4e** [reply]
"_The dependencies mentioned in this paragraph are referring to the PHP libraries & packages that Winter uses (i.e. Laravel, Symfony, Doctrine, etc - the underlying frameworks that Winter is built on top of)._"

I do need to install MariaDB myself, yes?

Should I run pkg_add -Uuv and syspatch and reboot, first?

Is this a good set of instructions to follow?
(https://techviewleo.com/install-and-configure-mariadb-database-on-openbsd/)

**Reply by bennothommo** [reply]
@kd4e you should be able to open a terminal and run `php -m` to show you the modules/extensions installed for PHP. As long as you have the extensions listed on the page @mjauvin mentioned (https://wintercms.com/docs/v1.2/docs/setup/installation), you should be good to go.

You will need a database - I'd recommend MySQL 8 over MariaDB these days, but it's entirely your choice. If you use either, you will need the `php-pdo-mysql` module to actually connect and function with those databases.

Re: `mbstring` - I've usually found (at least with Linux) that I had to specifically install that extension - it wasn't included by default. It might be different for OpenBSD however. Paraphrasing from https://www.php.net/manual/en/install.unix.openbsd.php, I think you would just need to run `pkg_add mbstring`.

I don't believe you need to `syspatch` and reboot unless you installed PHP FPM to connect to your webserver. You would generally just reboot the FPM service to ensure that any newly-installed extensions are loaded. Unfortunately, I have never used OpenBSD so can't help you on that one - rebooting may be the "cleaner" solution.

Hope that helps :)

**Reply by kd4e** [reply]
php-8.2 -m (forgetting to add the "-8.2" had me chasing my tail for a while ... sigh).
"bcmath, calendar, Core, ctype, curl, date, dom, exif, fileinfo, filter, ftp, gd, gettext, hash, iconv, json, libxml, mbstring, mysqlnd, openssl, pcre, PDO, Phar, posix, random, readline, Reflection, session, SimpleXML, sockets, sodium, SPL, standard, sysvmsg, sysvsem, sysvshm, tokenizer, xml, xmlreader, xmlwriter, Zend OPcache, zip, zlib [Zend Modules] Zend OPcache"

**Reply by bennothommo** [reply]
Great - looks like you have everything required. :)

**Reply by LukeTowers** [reply]
@kd4e are you able to write up a small section in the docs on the installation page for how to install the necessary requirements on OpenBSD and submit a PR?

**Reply by kd4e** [reply]
After I actually have WinterCMS working fully, yes.

**Reply by megido** [reply]
> After I actually have WinterCMS working fully, yes.

@kd4e can you post yours httpd.conf, I'm actually stuck on rewrites
